build-tools/packages/build-cli/src/codeCoverage/getCoverageMetrics.ts

```diff
@@ -8,46 +8,135 @@ import { Parser } from "xml2js";
 import type { CommandLogger } from "../logging.js";
 
 /**
- * The type for the coverage report, containing the line coverage and branch coverage(in percentage)
+ * The type for the coverage metric, containing the line, method and branch coverage(in percentage)
  */
 export interface CoverageMetric {
 	lineCoverage: number;
 	branchCoverage: number;
+	methodCoverage: number;
+}
+
+/**
+ * The type for the coverage metric for files in package, containing the line, method and branch coverage(in percentage)
+ */
+export interface CoverageMetricForPackages extends CoverageMetric {
+	filesCoverage: Map<string, CoverageMetric>;
 }
 
 interface XmlCoverageReportSchema {
 	coverage: {
-		packages: XmlCoverageReportSchemaForPackage[];
+		"$": {
+			"line-rate": string;
+			"branch-rate": string;
+		};
+		packages: [
+			{
+				package: XmlCoverageReportSchemaForPackage[];
+			},
+		];
 	};
 }
 
 interface XmlCoverageReportSchemaForPackage {
-	package: [
+	"$": {
+		name: string;
+		"line-rate": string;
+		"branch-rate": string;
+	};
+	classes: [
+		{
+			class?: XmlCoverageReportSchemaForFile[];
+		},
+	];
```

Comment:
The schema says classes but these seem to be files, correct? Can we add a comment here to make it clearer for people who read the code in the future?